opt/scopeserver/dataserver/modules/gserver/GServer.py

```diff
@@ -987,6 +973,46 @@ def loomUploaded(self, request, content):
         uploadedLooms[request.UUID].add(request.filename)
         return s_pb2.LoomUploadedReply()
 
+    def getGProfilerLink(self, request, context):
+        loom = self.lfh.get_loom(loom_file_path=request.loomFilePath)
+        cluster_marker_table = loom.get_cluster_marker_table(
+            clusteringID=request.clusteringID, clusterID=request.clusterID
+        )
+        if "avg_logFC" not in cluster_marker_table.columns:
+            error_message = f"Log fold change metric (avg_logFC) is not found in the cluster marker metric table."
+            logger.info(error_message)
+            return s_pb2.GProfilerLinkOutReply(error=s_pb2.ErrorReply(type="Value Error", message=error_message,))
+
+        def get_top_marker_genes_by(n, by="avg_logFC"):
+            top_genes_ranked = cluster_marker_table.sort_values(by=by, ascending=False).head(n=n).index.values
+            return np.insert(top_genes_ranked, 0, f">Top_{n}")
+
+        top_ranked_gene_lists = [
+            get_top_marker_genes_by(n=topNumFeaturesValue) for topNumFeaturesValue in sorted(request.topNumFeatures)
+        ]
```

Comment:
There could be a check here to confirm that all of the selected markers are in the same direction. 

It would actually be nice to add a separation between up and down regulated genes, much like you do for top100, top200 etc.

Just selecting the "top" X genes, could actually lead to a selection of genes with both positive and negative logFC values. From a biological view this could dilute the results as the upregulated and downregulated genes may have different functions.

Reply:
Also, we use `avg_logFC`, but not everyone might, using the frontend data and allowing sorting/filtering would prevent us limiting the looms that this can be used for.

Reply:
Further to this actually, this could also be useful for analysing target genes from regulons, but these don't have a log_fc